notify_socket.go

```diff
@@ -210,7 +210,7 @@ func sdNotifyBarrier(client *net.UnixConn) error {
 	var buf [1]byte
 	n, err := pipeR.Read(buf[:])
 	logrus.WithField("n", n).WithField("err", err).Info("Reading 1 byte returned")
-	if err == io.EOF {
+	if errors.Is(err, io.EOF) {
```

Comment:
Technically it's not needed; you can disable the linter warning using something like `//nolint:errorlint // comparison with io.EOF is legit.`

Reply:
Done.